### PR TITLE
LibJS: Implement nullish coalescing operator (??)

### PR DESCRIPTION
**Pull request by linusg**
I know, we're still in a phase where the parser can't even parse some of the basic stuff - but you gotta have [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) :D

### CODE REVIEW

**Comment by awesomekling** [comment]
Very cool Linus :smile:

File: Libraries/LibJS/Tests/logical-expressions-basic.js [hunk 1]
```diff
@@ -17,7 +17,6 @@ try {
     assert(("foo" && true) === true);
     assert((false && "bar") === false);
     assert((true && "bar") === "bar");
-    assert((null && true) === null);
```
Comment (awesomekling) [inline Libraries/LibJS/Tests/logical-expressions-basic.js:20]: This should still work, no? Why are we removing it?

File: Libraries/LibJS/Tests/logical-expressions-basic.js [hunk 2]
```diff
@@ -55,7 +54,6 @@ try {
     assert(("foo" || true) === "foo");
     assert((false || "bar") === "bar");
     assert((true || "bar") === true);
-    assert((null || true) === true);
```
Comment (awesomekling) [inline Libraries/LibJS/Tests/logical-expressions-basic.js:58]: This should still work, no? Why are we removing it?

File: Libraries/LibJS/Tests/logical-expressions-basic.js (hunk 1 above)
Comment (linusg) [reply Libraries/LibJS/Tests/logical-expressions-basic.js:20]: Oh no worries, it definitely still works. It's just a duplicated test :)

https://github.com/linusg/serenity/blob/d74ad81402c27113b2ed5586bc369703d6be61c4/Libraries/LibJS/Tests/logical-expressions-basic.js#L20

https://github.com/linusg/serenity/blob/d74ad81402c27113b2ed5586bc369703d6be61c4/Libraries/LibJS/Tests/logical-expressions-basic.js#L34

File: Libraries/LibJS/Tests/logical-expressions-basic.js (hunk 2 above)
Comment (linusg) [reply Libraries/LibJS/Tests/logical-expressions-basic.js:58]: Same here :)

File: Libraries/LibJS/Tests/logical-expressions-basic.js (hunk 1 above)
Comment (awesomekling) [reply Libraries/LibJS/Tests/logical-expressions-basic.js:20]: Aha! :)

**Pull request merged**